anta/runner.py

```diff
@@ -86,16 +100,22 @@ async def main(manager: ResultManager, inventory: AntaInventory, catalog: AntaCa
     coros = []
     # Using a set to avoid inserting duplicate tests
     tests_set: set[AntaTestRunner] = set()
-    for device in devices:
-        if tags:
-            # If there are CLI tags, only execute tests with matching tags
-            tests_set.update((test, device) for test in catalog.get_tests_by_tags(tags))
-        else:
-            # If there is no CLI tags, execute all tests without filters
-            tests_set.update((t, device) for t in catalog.tests if t.inputs.filters is None or t.inputs.filters.tags is None)
 
-            # Then add the tests with matching tags from device tags
-            tests_set.update((t, device) for t in catalog.get_tests_by_tags(device.tags))
+    for device in devices:
```

Comment:
Instead of modifying this for loop (which is already complex enough) could you:
1) modify AntaInventory.get_inventory() to be able to filter the inventory based on a device name (eventually a list of device names to support multiple --device argument to the CLI ? :) )
Better: create AntaInventory.get_devices() that can also support tags.
2) Implement AntaCatalog.get_tests() instead of AntaCatalog.get_tests_by_tags() that can get tests based on name and/or tags.

NB: (1) and (2) may not be as it is but I think we can have consistent methods on AntaCatalog and AntaInventory

3) filter devices and tests based on name before entering this loop.
4) only deal with tag management in this loop

Reply:
Code updated accordingly. For the moment, only 1 device and 1 test is supported by CLI, if change is OK, I will forward list support up to the CLI